### DISCUSSION TITLE
What is InLoopPollText commonly used for?

### DISCUSSION THREAD

**Discussion by jgarte** [qtile/qtile#4235] [Q&A]
Hi, what is [InLoopPollText](https://docs.qtile.org/en/stable/manual/howto/widget.html#inlooppolltext) commonly used for?

For example, the docs give an clear recommendation for how to use [_TextBox](https://docs.qtile.org/en/stable/manual/howto/widget.html#textbox):

> If all you are doing is displaying text then it's highly recommended that you use the base._TextBox superclass as this simplifies adding and updating text.

**Comment by elParaguayo** [comment]
You would use `InLoopPollText` if you have a function whose output you want to display in a widget. The function should exit quickly as otherwise it will block qtile's event loop. The function is called repeatedly with a defined interval.

If you just want to display text and update only on certain events then use the `TextBox`

**Reply by elParaguayo** [reply]
You'd only subclass `_TextBox` if you're creating a new widget. It's not intended to be used directly in your config.
